Objects/dictobject.c

```diff
@@ -3553,8 +3665,16 @@ dict_popitem_impl(PyDictObject *self)
         return NULL;
     }
     /* Convert split table to combined table */
-    if (self->ma_keys->dk_kind == DICT_KEYS_SPLIT) {
-        if (dictresize(interp, self, DK_LOG_SIZE(self->ma_keys), 1)) {
+    if (_PyDict_HasSplitTable(self)) {
+        PyDictKeysObject *keys = self->ma_keys;
+        dictkeys_incref(keys);
+        LOCK_KEYS(keys);
+
+        int status = dictresize(interp, self, DK_LOG_SIZE(self->ma_keys), 1);
+        dictkeys_decref(interp, keys);
```

Comment:
I think this `dictkeys_decref()` call would need to be after the `UNLOCK_KEYS`.

If I understand correctly, the extra incref/decref are because `dictresize` internally will decref the keys and we don't want them to be freed until after we unlock them.